tools/timemachine/internal/cmd/downloadHeight.go

```diff
@@ -0,0 +1,228 @@
+		if err != nil {
+			return errors.Wrapf(err, "Failed to convert input height: %s.", args[0])
+		}
+		if height == 0 {
+			return errors.New("input height cannot be 0.")
+		}
+		var (
+			heightDir string
+			destDir   = args[1]
+			bucket    = "blockchain-golden"
+			prefix    = "fullsync/mainnet/"
+			curr      uint64
+			wg        sync.WaitGroup
+		)
+
+		allObjNames, err := listFiles(bucket, "", "")
+		if err != nil {
+			return err
+		}
+		if len(allObjNames) == 0 {
+			return ErrNotExist
+		}
+		heiNames := make(map[string][]string)
+		for _, name := range allObjNames {
+			if strings.HasPrefix(name, prefix) {
+				s := strings.TrimPrefix(name, prefix)
+				if len(s) > 0 {
+					heiStr := s[:strings.Index(s, "/")]
+					_, ok := heiNames[heiStr]
+					if ok {
+						heiNames[heiStr] = append(heiNames[heiStr], name)
+					} else {
+						heiNames[heiStr] = make([]string, 0)
+					}
+
+					if heiStr != "latest" {
+						v := convertHeightStr(heiStr)
+						if v > curr {
+							curr = v
+						}
+					}
+				}
+			}
+		}
+		latest := curr + 250000
+
+		if height < 8000000 {
+			heightDir = "0m"
+		} else if height >= 8000000 && height < 12000000 {
+			heightDir = "8m"
+		} else if height >= 12000000 && height < 13000000 {
+			heightDir = "12m"
+		} else if height >= 13000000 && height < latest {
+			inter := height / 1000000
+			deci := height - inter
+			interStr := fmt.Sprintf("%dm", inter)
+			if deci < 250000 {
+				heightDir = interStr
+			} else if deci >= 250000 && deci < 500000 {
+				heightDir = interStr + "25"
+			} else if deci >= 500000 && deci < 750000 {
+				heightDir = interStr + "50"
+			} else if deci >= 750000 {
+				heightDir = interStr + "75"
+			}
+		} else if height >= latest && height < latest+250000 {
+			heightDir = "latest"
+		} else {
+			return errors.Errorf("input height: %s is larger than latest height: %v.", args[0], latest+250000)
+		}
+
+		cmd.Printf("download height dir is %s.\n", heightDir)
+
+		objs, ok := heiNames[heightDir]
+		if !ok {
+			return errors.Errorf("Failed to get height: %s", heightDir)
+		}
+		for _, obj := range objs {
+			wg.Add(1)
+			go func(obj string) {
+				defer wg.Done()
+				cmd.Printf("downlaoding height from %s.\n", obj)
+				if err := downloadFile(bucket, obj, filepath.Join(destDir, obj)); err != nil {
+					panic(errors.Wrapf(err, "Failed to downloadFile: %s.", obj))
+				}
+			}(obj)
+		}
+		wg.Wait()
+
+		cmd.Println("download height Successfully.")
+		return nil
+	},
+}
+
+// downloadFile downloads an object to a file.
+func downloadFile(bucket, object, destFileName string) error {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "Failed to storage.NewClient.")
+	}
+	defer client.Close()
+	client.SetRetry(storage.WithPolicy(storage.RetryAlways))
+
+	ctx, cancel := context.WithTimeout(ctx, GcpTimeout)
+	defer cancel()
+
+	if err = mkdirIfNotExist(filepath.Dir(destFileName)); err != nil {
```

Comment:
if destFileName does not exist, it is an error?

Reply:
if not exist, try to mkdir,  if error, return error.